components/delegations/columns.tsx

```diff
@@ -0,0 +1,265 @@
+
+import { type Row, type RowData, createColumnHelper, filterFns, sortingFns } from '@tanstack/react-table';
+import type { Delegation } from 'lib/delegate/DelegatePlatform';
+import { useTranslations } from 'next-intl';
+import { useMemo } from 'react';
+import ContractCell from './cells/ContractCell';
+import ControlsCell from './cells/ControlsCell';
+import DelegateCell from './cells/DelegateCell';
+import DelegationTypeCell from './cells/DelegationTypeCell';
+import DelegatorCell from './cells/DelegatorCell';
+import PlatformCell from './cells/PlatformCell';
+
+export enum ColumnId {
+  TYPE = 'Delegation Type',
+  DELEGATOR = 'Delegator',
+  DELEGATE = 'Delegate',
+  CONTRACT = 'Contract',
+  PLATFORM = 'Platform',
+  CONTROLS = 'Controls',
+}
+
+// Accessors for delegation data - these can be used for sorting and filtering
+export const accessors = {
+  type: (delegation: Delegation) => delegation.type,
+  delegator: (delegation: Delegation) => delegation.delegator,
+  delegate: (delegation: Delegation) => delegation.delegate,
+  contract: (delegation: Delegation) => delegation.contract,
+  platform: (delegation: Delegation) => delegation.platform,
+};
+
+// sorting functions
+export const customSortingFns = {
+  type: (rowA: Row<Delegation>, rowB: Row<Delegation>, columnId: string) => {
+    const typeOrder = {
+      ALL: 1,
+      CONTRACT: 2,
+      TOKEN: 3,
+      ERC721: 4,
+      ERC1155: 5,
+      NONE: 6,
+    };
+
+    const typeA = rowA.getValue(columnId) as string;
+    const typeB = rowB.getValue(columnId) as string;
+
+    const orderA = typeOrder[typeA as keyof typeof typeOrder] || 10;
+    const orderB = typeOrder[typeB as keyof typeof typeOrder] || 10;
+
+    return orderA - orderB;
+  },
+};
+
+// Custom filter functions
+export const customFilterFns = {
+  type: (row: Row<Delegation>, columnId: string, filterValues: string[]) => {
+    if (!filterValues.length) return true;
+    const results = filterValues.map((filterValue) => row.getValue(columnId) === filterValue);
+    return results.some((result) => result);
+  },
+  platform: (row: Row<Delegation>, columnId: string, filterValues: string[]) => {
+    if (!filterValues.length) return true;
+    const results = filterValues.map((filterValue) => {
+      return filterFns.includesString(row, columnId, filterValue, () => {});
+    });
+    return results.some((result) => result);
+  },
+};
+
+// This will extend the original TableMeta interface, not replace it
+declare module '@tanstack/table-core' {
+  interface TableMeta<TData extends RowData> {
+    onRevoke?: (delegation: Delegation) => void;
+  }
+}
+
+const columnHelper = createColumnHelper<Delegation>();
+
+// Function to create columns with translations
+const createColumns = (t: any, incoming: boolean) => {
+  // Common columns for both tables
+  const commonColumns = [
+    columnHelper.accessor(accessors.type, {
+      id: ColumnId.TYPE,
+      header: () => t('address.delegations.columns.type'),
```

Comment:
If you use `HeaderCell` here, then you don't need to wrap this in a wrapper for translations.